core-tests/shared/src/test/scala/zio/ZScheduleSpec.scala

```diff
@@ -148,50 +149,61 @@ object ZScheduleSpec
             val retried = (for {
               ref <- Ref.make(0)
               _   <- alwaysFail(ref).retry(Schedule.once)
-            } yield ()).foldM(
+            } yield ()).foldM[Any, Int, String](
               err => IO.succeed(err),
               _ => IO.succeed("A failure was expected")
             )
             assertM(retried, equalTo("Error: 2"))
           },
-          testM("for a given number of times") {
-            val retried  = retryCollect(IO.fail("Error"), Schedule.recurs(5))
-            val expected = (Left("Error"): Either[Any, Nothing], List(1, 2, 3, 4, 5, 6).map((Duration.Zero, _)))
-            assertM(retried, equalTo(expected))
-          },
           testM("for a given number of times with random jitter in (0, 1)") {
-            val schedule: ZSchedule[Random, Int, Int] = Schedule.recurs(5).delayed(_ => 500.millis).jittered
-            val scheduled: UIO[List[(Duration, Int)]] = schedule.run(List(1, 2, 3, 4, 5)).provide(TestRandom)
-            val expected                              = List(1, 2, 3, 4, 5).map((250.millis, _))
-            assertM(scheduled, equalTo(expected))
+            val schedule = ZSchedule.spaced(500.millis).jittered_[Clock, Clock](0, 1) { f => old =>
+              new Clock { val clock = f(old.clock) }
+            }
+            val scheduled = run(schedule >>> ZSchedule.elapsed)(List.fill(5)(()))
+            val expected  = List(0.millis, 250.millis, 500.millis, 750.millis, 1000.millis)
+            assertM(TestRandom.feedDoubles(0.5, 0.5, 0.5, 0.5, 0.5) *> scheduled, equalTo(expected))
           },
           testM("for a given number of times with random jitter in custom interval") {
-            val schedule: ZSchedule[Random, Int, Int] = Schedule.recurs(5).delayed(_ => 500.millis).jittered(2, 4)
-            val scheduled: UIO[List[(Duration, Int)]] = schedule.run(List(1, 2, 3, 4, 5)).provide(TestRandom)
-            val expected                              = List(1, 2, 3, 4, 5).map((1500.millis, _))
-            assertM(scheduled, equalTo(expected))
+            val schedule = ZSchedule.spaced(500.millis).jittered_[Clock, Clock](2, 4) { f => old =>
+              new Clock { val clock = f(old.clock) }
```

Comment:
These use the more painful `jittered_`

Reply:
I'll update the tests to use the new fiberTime that we have in TestClock. This will get cleaned up as part of it :+1: